app/main.py

```diff
@@ -1,107 +1,26 @@
-from app.knight_class import Knight
-from app.preparing_knights import battle_preparation
+from app.knight import Knight
+from app.config import KNIGHTS
+from app.arena import Arena
 
 
-KNIGHTS = {
-    "lancelot": {
-        "name": "Lancelot",
-        "power": 35,
-        "hp": 100,
-        "armour": [],
-        "weapon": {
-            "name": "Metal Sword",
-            "power": 50,
-        },
-        "potion": None,
-    },
-    "arthur": {
-        "name": "Arthur",
-        "power": 45,
-        "hp": 75,
-        "armour": [
-            {
-                "part": "helmet",
-                "protection": 15,
-            },
-            {
-                "part": "breastplate",
-                "protection": 20,
-            },
-            {
-                "part": "boots",
-                "protection": 10,
-            }
-        ],
-        "weapon": {
-            "name": "Two-handed Sword",
-            "power": 55,
-        },
-        "potion": None,
-    },
-    "mordred": {
-        "name": "Mordred",
-        "power": 30,
-        "hp": 90,
-        "armour": [
-            {
-                "part": "breastplate",
-                "protection": 15,
-            },
-            {
-                "part": "boots",
-                "protection": 10,
-            }
-        ],
-        "weapon": {
-            "name": "Poisoned Sword",
-            "power": 60,
-        },
-        "potion": {
-            "name": "Berserk",
-            "effect": {
-                "power": +15,
-                "hp": -5,
-                "protection": +10,
-            }
-        }
-    },
-    "red_knight": {
-        "name": "Red Knight",
-        "power": 40,
-        "hp": 70,
-        "armour": [
-            {
-                "part": "breastplate",
-                "protection": 25,
-            }
-        ],
-        "weapon": {
-            "name": "Sword",
-            "power": 45
-        },
-        "potion": {
-            "name": "Blessing",
-            "effect": {
-                "hp": +10,
-                "power": +5,
-            }
-        }
-    }
-}
-
-
-def battle(knights_config: dict) -> dict:
+def battle(
+        knights_config: dict,
+        pairs: list | tuple = None
+) -> dict:
     # BATTLE PREPARATIONS:
-    prepared_knights = battle_preparation(knights_config)
+    knights = Arena.equip_knights(knights_config)
+
+    battle_pairs = pairs or [
+        ["lancelot", "mordred"],
+        ["arthur", "red_knight"],
+    ]
```

Comment:
Best practice is to make predefined pairs, where you load these data, like from some config file or env.

Reply:
ok, I'll pay attention to this, but I won't rewrite it so as not to get too deep and bother you)